job_service/api/auth.py

```diff
@@ -0,0 +1,65 @@
+import logging
+from typing import Union
+
+import jwt
+from jwt import PyJWKClient
+from jwt.exceptions import (
+    InvalidSignatureError, ExpiredSignatureError,
+    InvalidAudienceError, DecodeError
+)
+
+from job_service.config import environment
+from job_service.exceptions import AuthError, NoUserError
+from job_service.model.job import UserInfo
+
+
+logger = logging.getLogger()
+jwks_client = PyJWKClient(environment.get('JWKS_URL'), lifespan=3000)
+
+
+def get_jwks_aud() -> str:
+    return 'datastore-qa' if environment.get('STACK') == 'qa' else 'datastore'
+
+
+def get_signing_key(jwt_token: str):
+    return jwks_client.get_signing_key_from_jwt(jwt_token).key
+
+
+def authorize_user(token: Union[str, None]) -> UserInfo:
```

Comment:
Maybe we should also check the `accreditation/role` here? Or should anyone be able to add new jobs?

Reply:
That is a good idea! I am on it 🏃🏻